src/map/homunculus.cpp

```diff
@@ -1483,216 +1437,555 @@ uint8 hom_get_intimacy_grade(struct homun_data *hd) {
-	db->gmin.vit = atoi(str[26]);
-	db->gmax.vit = atoi(str[27]);
-	db->gmin.int_= atoi(str[28]);
-	db->gmax.int_= atoi(str[29]);
-	db->gmin.dex = atoi(str[30]);
-	db->gmax.dex = atoi(str[31]);
-	db->gmin.luk = atoi(str[32]);
-	db->gmax.luk = atoi(str[33]);
-	//Evolution Min/Max HP, SP, str, agi, vit, int, dex, luk
-	db->emin.HP = atoi(str[34]);
-	db->emax.HP = atoi(str[35]);
-	db->emin.SP = atoi(str[36]);
-	db->emax.SP = atoi(str[37]);
-	db->emin.str = atoi(str[38]);
-	db->emax.str = atoi(str[39]);
-	db->emin.agi = atoi(str[40]);
-	db->emax.agi = atoi(str[41]);
-	db->emin.vit = atoi(str[42]);
-	db->emax.vit = atoi(str[43]);
-	db->emin.int_= atoi(str[44]);
-	db->emax.int_= atoi(str[45]);
-	db->emin.dex = atoi(str[46]);
-	db->emax.dex = atoi(str[47]);
-	db->emin.luk = atoi(str[48]);
-	db->emax.luk = atoi(str[49]);
-
-	//Check that the min/max values really are below the other one.
-	if(db->gmin.HP > db->gmax.HP)
-		db->gmin.HP = db->gmax.HP;
-	if(db->gmin.SP > db->gmax.SP)
-		db->gmin.SP = db->gmax.SP;
-	if(db->gmin.str > db->gmax.str)
-		db->gmin.str = db->gmax.str;
-	if(db->gmin.agi > db->gmax.agi)
-		db->gmin.agi = db->gmax.agi;
-	if(db->gmin.vit > db->gmax.vit)
-		db->gmin.vit = db->gmax.vit;
-	if(db->gmin.int_> db->gmax.int_)
-		db->gmin.int_= db->gmax.int_;
-	if(db->gmin.dex > db->gmax.dex)
-		db->gmin.dex = db->gmax.dex;
-	if(db->gmin.luk > db->gmax.luk)
-		db->gmin.luk = db->gmax.luk;
-
-	if(db->emin.HP > db->emax.HP)
-		db->emin.HP = db->emax.HP;
-	if(db->emin.SP > db->emax.SP)
-		db->emin.SP = db->emax.SP;
-	if(db->emin.str > db->emax.str)
-		db->emin.str = db->emax.str;
-	if(db->emin.agi > db->emax.agi)
-		db->emin.agi = db->emax.agi;
-	if(db->emin.vit > db->emax.vit)
-		db->emin.vit = db->emax.vit;
-	if(db->emin.int_> db->emax.int_)
-		db->emin.int_= db->emax.int_;
-	if(db->emin.dex > db->emax.dex)
-		db->emin.dex = db->emax.dex;
-	if(db->emin.luk > db->emax.luk)
-		db->emin.luk = db->emax.luk;
-
-	if (i >= homunculus_count)
-		homunculus_count++;
-	return true;
-}
 
-/**
-* Read homunculus db (check the files)
-*/
-void read_homunculusdb(void) {
-	uint8 i;
-	const char *filename[] = {
-		DBPATH"homunculus_db.txt",
-		DBIMPORT"/homunculus_db.txt",
-	};
-	homunculus_count = 0;
-	memset(homunculus_db,0,sizeof(homunculus_db));
-	for(i = 0; i<ARRAYLENGTH(filename); i++){
-		sv_readdb(db_path, filename[i], ',', 50, 50, MAX_HOMUNCULUS_CLASS, &read_homunculusdb_sub, i > 0);
+	if (this->nodeExists(node, "EvolutionClass")) {
+		std::string evo_class_name;
+
+		if (!this->asString(node, "EvolutionClass", evo_class_name))
+			return 0;
+
+		std::string evo_class_name_constant = "MER_" + evo_class_name;
+		int64 constant;
+
+		if (!script_get_constant(evo_class_name_constant.c_str(), &constant)) {
+			this->invalidWarning(node["EvolutionClass"], "Invalid homunculus Evolution Class %s, skipping.\n", evo_class_name.c_str());
+			return 0;
+		}
+
+		hom->evo_class = static_cast<int32>(constant);
+	} else {
+		if (!exists)
+			hom->evo_class = class_id;
```

Comment:
Because checks like `if(!hd->homunculusDB->evo_class || hd->homunculus.class_ == hd->homunculusDB->evo_class) {` seem more appropriate than just checking for 0! /s